src/pages/SignUp/SignUp.js

```diff
@@ -1,6 +1,156 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+// import Input from './Input';
 import './SignUp.scss';
 
-const SignUp = () => {};
+const SignUp = () => {
+  const [inputValues, setInputValues] = useState({
+    name: '',
+    email: '',
+    pw: '',
+    pwCorrect: '',
+  });
+
+  const { name, email, pw, pwCorrect } = inputValues;
+
+  const location = useLocation();
+  const CURRENT_DATA =
+    location.pathname === '/login' ? LOGIN_DATA : SIGNUP_DATA;
+
+  const handleInput = event => {
+    const { name, value } = event.target;
+    setInputValues({ ...inputValues, [name]: value });
+  };
+
+  const checkEmail = email.match('[a-zA-Z0-9]+[@][a-zA-Z0-9]+[.]+[a-z]{2,3}$');
+  const pwCheck = pw.match(
+    '^.*(?=^.{8,15}$)(?=.*d)(?=.*[a-zA-Z])(?=.*[!@#$%^&+=]).*$'
+  );
+
+  const conditions = {
+    name: name.length === 0 || name.length >= 2,
+    email: email.length === 0 || checkEmail,
+    pw: pw.length === 0 || pwCheck,
+    pwCorrect: pw === pwCorrect,
+  };
+
+  const signUp = event => {
+    if (name.length >= 2 && pwCheck && pwCorrect === pw && checkEmail) {
+      event.preventDefault();
+      fetch('http://10.58.52.150:8000/auth/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify({
+          name: `${name}`,
+          email: `${email}`,
+          password: `${pw}`,
+        }),
```

Comment:
```suggestion
        body: JSON.stringify({
          name: name,
          email: email,
          password: pw,
        }),
```
`${}` 를 사용하지 않아도 됩니다!

Reply:
```suggestion
        body: JSON.stringify({
          name,
          email,
          password: pw
        }),
```
객체단축속성명을 이용한다면 이렇게도 할 수 있겠네요!